gym_xiangqi/xiangqi_game.py

```diff
@@ -72,18 +82,68 @@ def on_event(self, event):
         if event.type == pygame.QUIT:
             self.running = False
 
+        elif event.type == pygame.MOUSEBUTTONDOWN:
+
+            if self.agent_turn:
+                # clicked: 1 not_clicked: 0
+                is_left_clicked = pygame.mouse.get_pressed()[0]
+
+                if is_left_clicked:
+                    # get clicked coordinate
+                    clicked_x, clicked_y = pygame.mouse.get_pos()
+                    clicked_coor = (clicked_x, clicked_y)
+
+                    # if piece is clicked, select it
+                    if self.find_target_piece(clicked_coor):
+                        pass
+
+                    else:
+                        # convert to real coordinate
+                        real_clicked_coor = self.to_real_coor(clicked_coor)
+                        is_valid_move = True  # set True for test
+
+                        # need to get this validity from env
+                        if is_valid_move and self.cur_selected is not None:
+                            enemies = self.enemy_piece[1:]
+                            enemy_coors = [piece.coor for piece in enemies]
+
+                            # if the coor is occupied by an enemy, kill it
+                            if real_clicked_coor in enemy_coors:
+                                self.kill_piece(real_clicked_coor)
+
+                            # fill the coordinate with the selected agent piece
+                            real_click_y = real_clicked_coor[1]
+                            real_click_x = real_clicked_coor[0]
+                            self.cur_selected.move(real_click_y, real_click_x)
+
+                            # reset counter after agent turn is over
+                            self.counter = COUNT
+
+                        # reset piece selection and end my turn
+                        self.cur_selected = None
+                        # self.agent_turn = False # commented for test
```

Comment:
I think we can possibly refactor the code to reduce a lot of nested if's, but it's definitely not something urgent.

Reply:
yea, I totally agree. I will optimize it as much as I can as we progress on this project